mitmproxy/websocket.py

```diff
@@ -0,0 +1,51 @@
+import time
+
+from typing import List
+
+from mitmproxy import flow
+from mitmproxy.types import serializable
+
+
+class WebSocketMessage(serializable.Serializable):
+
+    def __init__(self, from_client, content, timestamp=None):
+        self.content = content
+        self.from_client = from_client
+        self.timestamp = timestamp or time.time()
+
+    @classmethod
+    def from_state(cls, state):
+        return cls(*state)
+
+    def get_state(self):
+        return self.from_client, self.content, self.timestamp
+
+    def set_state(self, state):
+        self.from_client = state.pop("from_client")
+        self.content = state.pop("content")
+        self.timestamp = state.pop("timestamp")
+
+    def __repr__(self):
+        return "{direction} {content}".format(
+            direction="->" if self.from_client else "<-",
+            content=repr(self.content)
+        )
+
+
+class WebSocketFlow(flow.Flow):
```

Comment:
Right now this information is available in the `WebSocketLayer`. Do you want me to move it into the flow? Or is it fine if it stays in the layer?